src/utils/windowUtils.ts

```diff
@@ -41,6 +41,34 @@ export module Utils {
       }
     }
 
+    /**
+     * Sends a deprecation warning to the console. The warning includes the version number of the deprecation,
+     * the name of the function which was deprecated and an optional message.
+     *
+     * To be used in the first line of a deprecated method.
+     *
+     * @param {string} version The version when the tagged method became obsolete
+     * @param {string?} message Optional message to be shown with the warning
+     */
+    export function deprecated(version: string, message = "") {
+      var callingMethod = "";
+
+      try {
+        // Getting the name of the calling method through the stack trace.
+        callingMethod = (<any> new Error).stack
```

Comment:
Weird it seems that my comment got nuked somewhere...  Oh well.

Consider using this post to get the method name information instead of how you're doing it ccurrently https://stackoverflow.com/questions/280389/how-do-you-find-out-the-caller-function-in-javascript